src/MongoDB.Bson/ObjectModel/ObjectId.cs

```diff
@@ -31,7 +31,7 @@ public struct ObjectId : IComparable<ObjectId>, IEquatable<ObjectId>, IConvertib
     {
         // private static fields
         private static readonly ObjectId __emptyInstance = default(ObjectId);
-        private static readonly int __staticMachine = (GetMachineHash() + GetAppDomainId()) & 0x00ffffff;
+        private static readonly int __random = CalculateRandomValue();
```

Comment:
This value is not fully about `Random` logic because it doesn't take into account PId. But I'm ok to leave it as suggested

Reply:
also, it's 3 bytes size

Reply:
done